In [7]:
# !pip install openpyxl

In [8]:
import openpyxl
import pandas as pd
import numpy as np
import unicodedata

In [9]:
## COLUMNAS ESTANDAR
name_datos_base_cols = ['Idconce', 'Fecha', 'Nexpe', 'Kms', 'Sencillo', 'Indicencia',
                        'Gratuitos', 'Recargo', 'Bonobus', 'Abottes', 'Pmun', 'Otros', 'Valman',
                        'Total', 'Ingsencillo', 'Ingrecargo', 'Ingtotal', 'Fichero',
                        'Fecha Proc']
name_datos_via_cols = ['CONCESION', 'FECHA', 'NULINUSER', 'SUBLINEA', 'CODTIT', 'TITULO',
                       'IDCODIGOPERFIL', 'CODDES', 'DESCUENTO', 'VALIDACIONES', 'FICHERO','F_PROCESO']
name_datos_bit_cols = ['Operador', 'Concesion', 'Periodo', 'Codtit', 'Titulo', 'Subidos']
name_datos_vac_app_cols = ['Tipo', 'Empresa', 'Concesión/Linea', 'Titulo', 'Dscr Título', 'Fecha',
                            'Hora', 'Zona Validación', 'Validaciones']
name_datos_vac_val_cols = ['Empresa', 'Fecha', 'Concesion', 'L Gestra', 'Desclinea', 'P Gestra',
                            'Descparada', 'Corona', 'Codtit', 'Titulo', 'Validaciones']


In [10]:
datos_base = pd.read_excel("../2023-09/2023-09 DATOS_BASE.xlsx")
datos_via = pd.read_excel("../2023-09/2023-09 DATOS_VIA.xlsx", sheet_name=2)
datos_bit = pd.read_excel("../2023-09/2023-09 InformeBIT.xlsx", sheet_name=2)
datos_vac_app = pd.read_excel("../2023-09/2023-09Vac_app.xlsx", sheet_name=0)
datos_vac_val = pd.read_excel("../2023-09/2023-09Vac_val.xlsx", sheet_name=2)

# datos_base = pd.read_excel("./2023-08/2023-08DATOS_BASE_082023_EMTMADRID_CRTM_2023_10_6.xlsx")
# datos_via = pd.read_excel("./2023-08/2023-08DATOS_VIA_DETALLE___EMTMADRID_CRTM_2023_10_6.xlsx", sheet_name=2)
# datos_bit = pd.read_excel("./2023-08/2023-08 InformeBIT.xlsx", sheet_name=2)
# datos_vac_app = pd.read_excel("./2023-08/2023-08Vac_app.xlsx", sheet_name=0)
# datos_vac_val = pd.read_excel("./2023-08/2023-08 Vac_val.xlsx", sheet_name=0)

# datos_base = pd.read_excel("./2023-05/2023-05 DATOS_BASE.xlsx")
# datos_via = pd.read_excel("./2023-05/2023-05 DATOS_VIA_DETALLE.xlsx", sheet_name=2)
# datos_bit = pd.read_excel("./2023-05/2023-05 InformeBIT.xlsx", sheet_name=2)
# datos_vac_app = pd.read_excel("./2023-05/2023-05 Vac_app.xlsx", sheet_name=2)
# datos_vac_val = pd.read_excel("./2023-05/2023-05 Vac_Val.xlsx", sheet_name=1)

C:\Users\ray\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\openpyxl\styles\stylesheet.py:226: UserWarning: Workbook contains no default style, apply openpyxl's default
  warn("Workbook contains no default style, apply openpyxl's default")


In [11]:
print("BASE ",datos_base.shape)
print("VIA ",datos_via.shape)
print("BIT ",datos_bit.shape)
print("VAC_APP ",datos_vac_app.shape)
print("VAC_VAL ",datos_vac_val.shape)

BASE  (42, 19)
VIA  (760665, 12)
BIT  (3002, 6)
VAC_APP  (149352, 9)
VAC_VAL  (21172, 11)


In [12]:
datos_bit

Operador Concesion  Periodo  Codtit  \
0                         EMT MADRID        97  09/2023    8196   
1                         EMT MADRID       NaN  09/2023    4179   
2            NEXCONTINENTAL HOLDINGS  URCM-148  09/2023    8213   
3     AVANZA MOVILIDAD INTEGRAL S.L.   VCM-601  09/2023    4099   
4            NEXCONTINENTAL HOLDINGS   VCM-202  09/2023    4181   
...                              ...       ...      ...     ...   
2997         NEXCONTINENTAL HOLDINGS         -  09/2023    9482   
2998                 METRO DE MADRID       NaN  09/2023    9513   
2999                 METRO DE MADRID       NaN  09/2023    4691   
3000                          T.F.M.       NaN  09/2023    4131   
3001                 CERCANIAS RENFE       NaN  09/2023    4182   

                            Titulo  Subidos  
0               METROBUS 10 VIAJES  3824159  
1         ABONO ANUAL TERCERA EDAD      647  
2                BONOBUS URBANO B2    35319  
3                  30 DIAS ZONA B2   365982  
4     ABONO 30 DIAS JOVEN T. PLANA   409492  
...                            ...      ...  
2997           SENCILLO ZONA B2-B3        7  
2998            SENCILLO COMBINADO    53626  
2999      ABONO ANUAL TERCERA EDAD        2  
3000             TURISTICO T 1 DIA        8  
3001    ABONO ANUAL JOVEN T. PLANA      965  

[3002 rows x 6 columns]

In [13]:
## NOMBRES COLUMNAS ARCHIVO ACTUAL
print("datos_base: ",datos_base.columns, datos_base.shape)
print("datos_via: ",datos_via.columns, datos_via.shape)
print("datos_bit: ",datos_bit.columns, datos_bit.shape)
print("datos_vac_app: ",datos_vac_app.columns, datos_vac_app.shape)
print("datos_vac_val: ",datos_vac_val.columns, datos_vac_val.shape)
## NOMBRES COLUMNAS ESTANDAR
a,b = 'ÁÉÍÓÚÄËÏÖÜ', 'AEIOUAEIOU'
trans = str.maketrans(a,b)
name_datos_via_cols_OK = [column_name.upper().translate(trans) for column_name in name_datos_via_cols]
name_datos_bit_cols_OK = [column_name.upper().translate(trans) for column_name in name_datos_bit_cols]
name_datos_vac_app_cols_OK = [column_name.upper().translate(trans) for column_name in name_datos_vac_app_cols]
name_datos_vac_val_cols_OK = [column_name.upper().translate(trans) for column_name in name_datos_vac_val_cols]

print("name_datos_via_cols_OK: ",name_datos_via_cols_OK)
print("name_datos_bit_cols_OK: ",name_datos_bit_cols_OK)
print("name_datos_vac_app_cols_OK: ",name_datos_vac_app_cols_OK)
print("name_datos_vac_val_cols_OK: ",name_datos_vac_val_cols_OK)

datos_base:  Index(['Idconce', 'Fecha', 'Nexpe', 'Kms', 'Sencillo', 'Indicencia',
       'Gratuitos', 'Recargo', 'Bonobus', 'Abottes', 'Pmun', 'Otros', 'Valman',
       'Total', 'Ingsencillo', 'Ingrecargo', 'Ingtotal', 'Fichero',
       'Fecha Proc'],
      dtype='object') (42, 19)
datos_via:  Index(['CONCESION', 'FECHA', 'NULINUSER', 'SUBLINEA', 'CODTIT', 'TITULO',
       'IDCODIGOPERFIL', 'CODDES', 'DESCUENTO', 'VALIDACIONES', 'FICHERO',
       'F_PROCESO'],
      dtype='object') (760665, 12)
datos_bit:  Index(['Operador', 'Concesion', 'Periodo', 'Codtit', 'Titulo', 'Subidos'], dtype='object') (3002, 6)
datos_vac_app:  Index(['Tipo', 'Empresa', 'Concesión/Linea', 'Titulo', 'Dscr Título', 'Fecha',
       'Hora', 'Zona Validación', 'Validaciones'],
      dtype='object') (149352, 9)
datos_vac_val:  Index(['Empresa', 'Fecha', 'Concesion', 'L Gestra', 'Desclinea', 'P Gestra',
       'Descparada', 'Corona', 'Codtit', 'Titulo', 'Validaciones'],
      dtype='object') (21172, 11)
name_datos_v

## Comprobamos los nombres de las columnas
Para procesarse tienen que coincidir con los estandar (como los del mes 08/2023).

- datos_base:  Index(['Idconce', 'Fecha', 'Nexpe', 'Kms', 'Sencillo', 'Indicencia',
       'Gratuitos', 'Recargo', 'Bonobus', 'Abottes', 'Pmun', 'Otros', 'Valman',
       'Total', 'Ingsencillo', 'Ingrecargo', 'Ingtotal', 'Fichero',
       'Fecha Proc'],
      dtype='object') (42, 19)
- datos_via:  Index(['CONCESION', 'FECHA', 'NULINUSER', 'SUBLINEA', 'CODTIT', 'TITULO',
       'IDCODIGOPERFIL', 'CODDES', 'DESCUENTO', 'VALIDACIONES', 'FICHERO',
       'F_PROCESO'],
      dtype='object') (760665, 12)
- datos_bit:  Index(['Operador', 'Concesion', 'Periodo', 'Codtit', 'Titulo', 'Subidos'], dtype='object') (3002, 6)
- datos_vac_app:  Index(['Tipo', 'Empresa', 'Concesión/Linea', 'Titulo', 'Dscr Título', 'Fecha',
       'Hora', 'Zona Validación', 'Validaciones'],
      dtype='object') (149352, 9)
datos_vac_val:  Index(['Empresa', 'Fecha', 'Concesion', 'L Gestra', 'Desclinea', 'P Gestra',
       'Descparada', 'Corona', 'Codtit', 'Titulo', 'Validaciones'],
      dtype='object') (21172, 11)

###### Comprobacion previa

In [14]:
def estandarizar_nombres_columnas(datos_via, datos_bit, datos_vac_app, datos_vac_val):
    ## pasarlos a mayusculas y quitar acentos y espacios laterales. 
    ## sustituir espacios intermedios por _
    a,b = 'ÁÉÍÓÚÄËÏÖÜ', 'AEIOUAEIOU'
    trans = str.maketrans(a,b)
    datos_via_cols = [column_name.upper().translate(trans) for column_name in datos_via.columns]
    datos_bit_cols = [column_name.upper().translate(trans) for column_name in datos_bit.columns]
    datos_vac_app_cols = [column_name.upper().translate(trans) for column_name in datos_vac_app.columns]
    datos_vac_val_cols = [column_name.upper().translate(trans) for column_name in datos_vac_val.columns]
    return datos_via_cols, datos_bit_cols, datos_vac_app_cols, datos_vac_val_cols

def comprobar_nombres_columnas(datos_via, datos_bit, datos_vac_app, datos_vac_val):
    # name_datos_via_cols_OK, name_datos_bit_cols_OK, name_datos_vac_app_cols_OK, name_datos_vac_val_cols_OK
    datos_via_cols, datos_bit_cols, datos_vac_app_cols, datos_vac_val_cols = estandarizar_nombres_columnas(datos_via, datos_bit, datos_vac_app, datos_vac_val)
    print("COMPROBANDO LAS DIFERENCIAS ENTRE LAS COLUMNAS: ")
    print("__________________________________________________________\n")
    cols_data_via = [col for col in datos_via_cols if col not in name_datos_via_cols_OK]
    if cols_data_via == []:
        print("CORRECTO, LOS CAMPOS DE LAS COLUMNAS DEL datos_via COINCIDEN.")
    else:
        print("CORREGIR CAMPOS datos_via: ")
        print("Su fichero contiene las columnas: ", datos_via.columns)
        print("El fichero deseado contiene las columnas: ", name_datos_via_cols)
        print("Corrija las siguientes columnas: ", cols_data_via)
    print("__________________________________________________________\n")
    cols_data_bit = [col for col in datos_bit_cols if col not in name_datos_bit_cols_OK]
    if cols_data_bit == []:
        print("CORRECTO, LOS CAMPOS DE LAS COLUMNAS DEL datos_bit COINCIDEN.")
    else:
        print("CORREGIR CAMPOS cols_data_bit: ")
        print("Su fichero contiene las columnas: ", datos_bit.columns)
        print("El fichero deseado contiene las columnas: ", name_datos_bit_cols)
        print("Corrija las siguientes columnas: ", cols_data_bit)
    print("__________________________________________________________\n")
    cols_data_vac_app = [col for col in datos_vac_app_cols if col not in name_datos_vac_app_cols_OK]
    if cols_data_vac_app == []:
        print("CORRECTO, LOS CAMPOS DE LAS COLUMNAS DEL datos_vac_app COINCIDEN.")
    else:
        print("CORREGIR CAMPOS datos_vac_app: ")
        print("Su fichero contiene las columnas: ", datos_vac_app.columns)
        print("El fichero deseado contiene las columnas: ", name_datos_vac_app_cols)
        print("Corrija las siguientes columnas: ", cols_data_vac_app)
    print("__________________________________________________________\n")
    cols_data_vac_val = [col for col in datos_vac_val_cols if col not in name_datos_vac_val_cols_OK]
    if cols_data_vac_val == []:
        print("CORRECTO, LOS CAMPOS DE LAS COLUMNAS DEL datos_vac_val COINCIDEN.")
    else:
        print("CORREGIR CAMPOS datos_vac_val: ")
        print("Su fichero contiene las columnas: ", datos_vac_val.columns)
        print("El fichero deseado contiene las columnas: ", name_datos_vac_val_cols)
        print("Corrija las siguientes columnas: ", cols_data_vac_val)


In [15]:
comprobar_nombres_columnas(datos_via, datos_bit, datos_vac_app, datos_vac_val)

COMPROBANDO LAS DIFERENCIAS ENTRE LAS COLUMNAS: 
__________________________________________________________

CORRECTO, LOS CAMPOS DE LAS COLUMNAS DEL datos_via COINCIDEN.
__________________________________________________________

CORRECTO, LOS CAMPOS DE LAS COLUMNAS DEL datos_bit COINCIDEN.
__________________________________________________________

CORRECTO, LOS CAMPOS DE LAS COLUMNAS DEL datos_vac_app COINCIDEN.
__________________________________________________________

CORRECTO, LOS CAMPOS DE LAS COLUMNAS DEL datos_vac_val COINCIDEN.


## Procesamos el VAC

#### Procesamos el datos_vac_val

In [16]:
day_datos_vac_val_export = datos_vac_val.groupby(["Concesion", "Fecha", "Codtit"])["Validaciones"].sum().reset_index(name="Total_VAC")

In [17]:
## Comprobacion de que funciona
# '''
# ['Empresa', 'Fecha', 'Concesion', 'L Gestra', 'Desclinea', 'P Gestra',
#        'Descparada', 'Corona', 'Codtit', 'Titulo', 'Validaciones']
# '''
# datos_vac_val[(datos_vac_val["Concesion"]=="V5805") & (datos_vac_val["Fecha"]=="2023-09-01") & (datos_vac_val["Codtit"]==4104)]

# day_datos_vac_val_export

#### Procesamos el datos_vac_app

In [18]:
day_datos_vac_app_export = datos_vac_app.groupby(["Concesión/Linea", "Fecha", "Titulo"])["Validaciones"].sum().reset_index(name="Total_VAC")
day_datos_vac_app_export.rename({"Concesión/Linea":"Concesion", "Titulo":"Codtit"}, axis=1, inplace=True)

In [19]:
## Comprobacion de que funciona
# '''
# ['Empresa', 'Fecha', 'Concesion', 'L Gestra', 'Desclinea', 'P Gestra',
#        'Descparada', 'Corona', 'Codtit', 'Titulo', 'Validaciones']
# '''
# datos_vac_val[(datos_vac_val["Concesion"]=="V5805") & (datos_vac_val["Fecha"]=="2023-09-01") & (datos_vac_val["Codtit"]==4104)]

day_datos_vac_app_export

Concesion      Fecha  Codtit  Total_VAC
0      VAC-051 2023-09-01    4105         31
1      VAC-051 2023-09-01    4181         78
2      VAC-051 2023-09-02    4105         15
3      VAC-051 2023-09-02    4181         68
4      VAC-051 2023-09-03    4105         18
...        ...        ...     ...        ...
1791   VAC-249 2023-09-26    4147          1
1792   VAC-249 2023-09-26    4181          3
1793   VAC-249 2023-09-26    4706          3
1794   VAC-249 2023-09-27    4181          1
1795   VAC-249 2023-09-30    4181          3

[1796 rows x 4 columns]

In [20]:
print("columnas datos_vac_app",day_datos_vac_app_export.columns)
print("columnas datos_vac_val",day_datos_vac_val_export.columns)

columnas datos_vac_app Index(['Concesion', 'Fecha', 'Codtit', 'Total_VAC'], dtype='object')
columnas datos_vac_val Index(['Concesion', 'Fecha', 'Codtit', 'Total_VAC'], dtype='object')


In [21]:
day_datos_vac_export = pd.concat([day_datos_vac_val_export, day_datos_vac_app_export], axis=0)
# day_datos_vac_export

In [22]:
month_datos_vac_export = day_datos_vac_export
## sustituyo la fecha a mes
if month_datos_vac_export.Fecha.dtypes == "datetime64[ns]":
    month_datos_vac_export.Fecha = day_datos_vac_export.Fecha.dt.to_period("M")
## agrupo por mes y hago la suma total de los totales

In [23]:
month_datos_vac_export = month_datos_vac_export.groupby(["Concesion", "Fecha"])["Total_VAC"].sum().reset_index(name="Total_VAC")
# month_datos_vac_export

In [24]:
month_datos_vac_export

Concesion    Fecha  Total_VAC
0      V5805  2023-09       9970
1    VAC-023  2023-09     105980
2    VAC-051  2023-09       4694
3    VAC-063  2023-09      37191
4    VAC-082  2023-09      25913
5    VAC-087  2023-09      33305
6    VAC-093  2023-09       1783
7    VAC-152  2023-09      15312
8    VAC-158  2023-09      56814
9    VAC-221  2023-09        158
10   VAC-226  2023-09        925
11   VAC-231  2023-09      66827
12   VAC-243  2023-09      53472
13   VAC-249  2023-09         95

In [25]:
# datos_vac_export.to_excel("./2023-09/2023-09Vac_export.xlsx", index=False)

## Procesamos el BIT

In [26]:
month_datos_bit_export = datos_bit.groupby(["Concesion", "Periodo", "Codtit"])["Subidos"].sum().reset_index(name="Total_BIT")

In [27]:
# month_datos_bit_export

In [28]:
month_datos_bit_export.Periodo = pd.to_datetime(month_datos_bit_export.Periodo, format="%m/%Y")
if month_datos_bit_export.Periodo.dtypes == "datetime64[ns]":
    month_datos_bit_export.Periodo = month_datos_bit_export.Periodo.dt.to_period("M")
month_datos_bit_export.columns

Index(['Concesion', 'Periodo', 'Codtit', 'Total_BIT'], dtype='object')

In [29]:
month_datos_bit_export = month_datos_bit_export.groupby(["Concesion", "Periodo"])["Total_BIT"].sum().reset_index(name="Total_BIT")
# month_datos_bit_export

## Procesamos el VIA

In [30]:
day_datos_via_export = datos_via.groupby(["CONCESION", "FECHA", "CODTIT"])["VALIDACIONES"].sum().reset_index(name="Total_VIA")

In [31]:
day_datos_via_export.FECHA = pd.to_datetime(day_datos_via_export.FECHA, format="%d/%m/%Y")
month_datos_via_export = day_datos_via_export
## sustituyo la fecha a mes
if month_datos_via_export.FECHA.dtypes == "datetime64[ns]":
    month_datos_via_export.FECHA = pd.to_datetime(month_datos_via_export.FECHA)
    month_datos_via_export.FECHA = day_datos_via_export.FECHA.dt.to_period("M")
## agrupo por mes y hago la suma total de los totales

In [32]:
month_datos_via_export = month_datos_via_export.groupby(["CONCESION", "FECHA"])["Total_VIA"].sum().reset_index(name="Total_VIA")
# month_datos_via_export

## Datos Base

In [33]:
datos_base.columns

Index(['Idconce', 'Fecha', 'Nexpe', 'Kms', 'Sencillo', 'Indicencia',
       'Gratuitos', 'Recargo', 'Bonobus', 'Abottes', 'Pmun', 'Otros', 'Valman',
       'Total', 'Ingsencillo', 'Ingrecargo', 'Ingtotal', 'Fichero',
       'Fecha Proc'],
      dtype='object')

In [34]:
datos_base.rename(columns={"Idconce":"Concesion","Fecha":"Mes"}, inplace=True, errors="raise")
datos_base_ok = datos_base[["Concesion","Mes","Total"]]
datos_base_ok.rename(columns={"Total":"Total_BASE"}, inplace=True)


C:\Users\ray\AppData\Local\Temp\ipykernel_23900\1141183840.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  datos_base_ok.rename(columns={"Total":"Total_BASE"}, inplace=True)


In [35]:
datos_base_ok

Concesion      Mes  Total_BASE
0   URCM-005  09/2023     1054290
1   URCM-013  09/2023      252318
2   URCM-014  09/2023      117990
3   URCM-148  09/2023      505390
4   URCM-152  09/2023       55576
5   URCM-161  09/2023      413835
6    VAC-023  09/2023      107516
7    VAC-082  09/2023       26242
8    VAC-226  09/2023        1348
9    VAC-231  09/2023       66699
10   VAC-243  09/2023       53506
11   VCM-101  09/2023     1317462
12   VCM-102  09/2023      136024
13   VCM-103  09/2023      391041
14   VCM-200  09/2023      248875
15   VCM-201  09/2023      939016
16   VCM-202  09/2023     1202659
17   VCM-203  09/2023      472570
18   VCM-204  09/2023       37250
19   VCM-301  09/2023      999588
20   VCM-302  09/2023      332836
21   VCM-303  09/2023       88289
22   VCM-401  09/2023      744572
23   VCM-402  09/2023      947817
24   VCM-403  09/2023      738178
25   VCM-404  09/2023     2017154
26   VCM-500  09/2023      221191
27   VCM-501  09/2023     3235216
28   VCM-502  09/2023      497578
29   VCM-503  09/2023      114980
30   VCM-504  09/2023      125523
31   VCM-600  09/2023       83467
32   VCM-601  09/2023     2013677
33   VCM-602  09/2023     1233735
34   VCM-603  09/2023      233821
35   VCM-604  09/2023      848406
36   VCM-605  09/2023      454438
37   VCM-606  09/2023      460936
38   VCM-607  09/2023      371511
39   VCM-701  09/2023      499860
40   VCM-702  09/2023      613866
41     V5805  09/2023       15222

In [36]:
datos_base_ok.Mes = pd.to_datetime(datos_base_ok.Mes, format="%m/%Y")
if datos_base_ok.Mes.dtypes == "datetime64[ns]":
    datos_base_ok.Mes = pd.to_datetime(datos_base_ok.Mes)
    datos_base_ok.Mes = datos_base_ok.Mes.dt.to_period("M")

C:\Users\ray\AppData\Local\Temp\ipykernel_23900\3157417470.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  datos_base_ok.Mes = pd.to_datetime(datos_base_ok.Mes, format="%m/%Y")
C:\Users\ray\AppData\Local\Temp\ipykernel_23900\3157417470.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  datos_base_ok.Mes = pd.to_datetime(datos_base_ok.Mes)
C:\Users\ray\AppData\Local\Temp\ipykernel_23900\3157417470.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using

## Agrupo todas las columnas por concesion y fecha

In [37]:
## renombramos todas las columnas del dataframe
month_datos_via_export.rename({"CONCESION":"Concesion", "FECHA":"Mes", "Total":"Total"}, axis=1, inplace=True)
month_datos_bit_export.rename({"CONCESION":"Concesion", "Periodo":"Mes", "Total":"Total"}, axis=1, inplace=True)
month_datos_vac_export.rename({"CONCESION":"Concesion", "Fecha":"Mes", "Total":"Total"}, axis=1, inplace=True)

print("columnas datos_via: ",month_datos_via_export.columns)
print("columnas datos_bit: ",month_datos_bit_export.columns)
print("columnas datos_vac: ",month_datos_vac_export.columns)

# print("columnas datos_base: ",datos_base.columns)

columnas datos_via:  Index(['Concesion', 'Mes', 'Total_VIA'], dtype='object')
columnas datos_bit:  Index(['Concesion', 'Mes', 'Total_BIT'], dtype='object')
columnas datos_vac:  Index(['Concesion', 'Mes', 'Total_VAC'], dtype='object')


In [38]:
print("BIT_TIPO_MES: ",month_datos_bit_export.Mes.dtypes)
print("BIT_TIPO_VIA: ",month_datos_via_export.Mes.dtypes)
print("BIT_TIPO_VAC: ",month_datos_vac_export.Mes.dtypes)

BIT_TIPO_MES:  period[M]
BIT_TIPO_VIA:  period[M]
BIT_TIPO_VAC:  period[M]


In [39]:
lista_concat = ['Concesion', 'Mes']

lista_totales_union_INTERURBANOS = pd.merge(pd.merge(month_datos_via_export, 
    month_datos_bit_export, on=lista_concat, how="left")
    , datos_base_ok, on=lista_concat, how="left")

lista_totales_union_INTERURBANOS.rename({"Total_x":"Total_via", "Total_y":"Total_bit", "Total":"Total_BASE"}, axis=1, inplace=True)
lista_totales_union_INTERURBANOS.drop(["Mes"], axis=1, inplace=True)
# lista_totales_union_INTERURBANOS = lista_totales_union_INTERURBANOS.applymap(lambda x: f'{x:.2f}' if pd.notna(x) and isinstance(x, (float, int)) else x)

lista_totales_union_VACs = pd.merge(pd.merge(pd.merge(month_datos_vac_export, month_datos_bit_export, on=lista_concat, how="left"), datos_base_ok, on=lista_concat, how="left"), month_datos_via_export, on=lista_concat, how="left")
lista_totales_union_VACs.rename({"Total_x":"Total_via", "Total_y":"Total_bit", "Total":"Total_BASE"}, axis=1, inplace=True)
lista_totales_union_VACs.drop(["Mes"], axis=1, inplace=True)
# lista_totales_union_VACs = lista_totales_union_VACs.applymap(lambda x: f'{x:.2f}' if pd.notna(x) and isinstance(x, (float, int)) else x)

In [40]:
lista_totales_union_INTERURBANOS = lista_totales_union_INTERURBANOS[(lista_totales_union_INTERURBANOS.Concesion != "-") & (lista_totales_union_INTERURBANOS.Concesion != "97")]

In [41]:
lista_vacs_interurbano = ['V5805', 'VAC-023', 'VAC-051', 'VAC-063', 'VAC-082', 'VAC-087', 'VAC-093', 'VAC-152', 'VAC-158', 'VAC-221', 'VAC-226', 'VAC-231', 'VAC-243', 'VAC-249']

interurbanos_VAC = lista_totales_union_INTERURBANOS[lista_totales_union_INTERURBANOS['Concesion'].isin(lista_vacs_interurbano)]
interurbanos_VCM = lista_totales_union_INTERURBANOS[~lista_totales_union_INTERURBANOS['Concesion'].isin(lista_vacs_interurbano)]


In [36]:
lista_totales_union_VACs = lista_totales_union_VACs[(lista_totales_union_VACs.Concesion != "-") & (lista_totales_union_VACs.Concesion != "97")]

In [37]:
## generamos dos listas, una para interurbanos y otra para VACs
lista_totales_union_INTERURBANOS.to_excel("./pruebas_out_app/interurbanos.xlsx", index=None)
lista_totales_union_VACs.to_excel("./pruebas_out_app/vacs.xlsx", index=None)
interurbanos_VAC.to_excel("./pruebas_out_app/interurbanos_VAC.xlsx", index=None)
interurbanos_VCM.to_excel("./pruebas_out_app/interurbanos_VCM.xlsx", index=None)

In [38]:
lista_totales_union_VACs.columns

Index(['Concesion', 'Total_VAC', 'Total_BIT', 'Total_BASE', 'Total_VIA'], dtype='object')